### PR TITLE
Itsadssd 29483 color consolidation

### PR DESCRIPTION
**Pull request by uqtgeor4**
Reference 

UQ Colors
https://www.figma.com/file/Li7SZkmaLgw8ZdiHje2q7x/Future-Students---Design-System?node-id=1283%3A0
https://uq.edu.au/

Tint/Shade (for neutral tones)
https://maketintsandshades.com/

Lighten/Darken (for UQ primary, secondary colours)
http://scg.ar-ch.org/

### CODE REVIEW

File: packages/core/src/scss/_partials/_colour-constants.scss [hunk 1]
```diff
@@ -55,17 +79,19 @@ $link-hover: #005ea5;
 $link-active: $link-hover;
 
 // Alerts and Status - Dark
-$alert: $uq-red-dark20;
+$info: $uq-blue;
 $success: $uq-green-dark10;
+$warning: $uq-yellow;
```
Comment (uqmchu) [inline packages/core/src/scss/_partials/_colour-constants.scss:84]: This also fails contrast requirements with white text.

File: packages/core/src/scss/_partials/_colour-constants.scss [hunk 2]
```diff
@@ -1,15 +1,26 @@
+/* Variants of brand colours created via below SASS functions
```
Comment (uqmchu) [inline packages/core/src/scss/_partials/_colour-constants.scss:1]: 👍

File: packages/core/src/scss/_partials/_colour-constants.scss [hunk 3]
```diff
@@ -1,15 +1,26 @@
...
+   Primary, Secondary colour : Lighten/Darken (% increments of 10)
+   Neutral colour : Tint/Shade (% increments of 20)
+*/
+
+// Function to Tint a color
+@function tint($color, $percentage) {
+    @return mix(white, $color, $percentage);
```
Comment (uqmchu) [inline packages/core/src/scss/_partials/_colour-constants.scss:8]: If you had just swapped the order of `white` and `$color` arguments in the mix function, you wouldn't have had to go through and rename all the colour variables for the neutrals. Although, I can appreciate why you did it.

File: packages/storybook-html/src/components/core/colour-apply.story.mdx [hunk 4]
```diff
@@ -177,9 +177,15 @@ Backgrounds are used to differentiate parts of an interface, and to create conta
 
 See for example: Sections, Cards, Table -- *add links eventually*
 
+The SASS function 'mix' is used to create variants for UQ neutral tones only. In all other UQ primary or secondary colours, the SASS function 'lighten' or 'darken' is used to create it's variants.
```
Comment (uqmchu) [inline packages/storybook-html/src/components/core/colour-apply.story.mdx:180]: The audience doesn't need to know these details and it distracts from the colour usage guidance.

File: packages/storybook-html/src/components/core/colour-apply.story.mdx [hunk 5]
```diff
@@ -177,9 +177,15 @@ Backgrounds are used to differentiate parts of an interface, and to create conta
...
 
 ### Light backgrounds
 
+The light grey colour is a lighter variant of UQ Neutral colour produced by SASS function 'mix',
```
Comment (uqmchu) [inline packages/storybook-html/src/components/core/colour-apply.story.mdx:184]: Also remove this.

File: packages/storybook-html/src/components/core/colour-apply.story.mdx [hunk 6]
```diff
@@ -249,6 +255,8 @@ These colours can only be used for components which alert a user or display a st
 
 These can be used as background colours to default text, e.g. alerts.
 
+These light colours are lighter variants of its corresponding UQ brand colours produced by SASS function 'lighten', e.g. <code>$info-light: lighten($uq-blue, 40%);</code>
```
Comment (uqmchu) [inline packages/storybook-html/src/components/core/colour-apply.story.mdx:258]: Remove

File: packages/storybook-html/src/components/core/colour-apply.story.mdx [hunk 7]
```diff
@@ -294,6 +302,8 @@ These can be used as background colours to default text, e.g. alerts.
 
 These can be used on a white background or as background colours to white text, e.g. form validation errors.
 
+Some of these dark colours are darker variants of its corresponding UQ brand colour produced by SASS function 'darken', e.g. <code>$alert: darken($uq-red, 20%);</code>
```
Comment (uqmchu) [inline packages/storybook-html/src/components/core/colour-apply.story.mdx:305]: Remove

File: packages/storybook-html/src/components/core/colour-base.story.mdx [hunk 8]
```diff
@@ -35,21 +35,21 @@ These are the core brand colours for the University of Queensland. Refer to the
     </tr>
     <tr  className="uqds-table__row">
       <td className="uqds-table__item"><div className="uqds-colour__palette uqds-colour--purple"></div></td>
-      <td className="uqds-table__item" data-title="Name:">UQ purple</td>
+      <td className="uqds-table__item" data-title="Name:">UQ Purple</td>
```
Comment (uqmchu) [inline packages/storybook-html/src/components/core/colour-base.story.mdx:38]: Revert the naming. Should be sentence casing. UQ purple is not really a proper noun.

File: packages/storybook-html/src/components/core/colour-base.story.mdx [hunk 9]
```diff
@@ -150,7 +150,7 @@ Neutral tones can be used as a background colour.
     </tr>
     <tr className="uqds-table__row">
       <td className="uqds-table__item"><div className="uqds-colour__palette uqds-colour--darkgrey"></div></td>
-      <td className="uqds-table__item" data-title="Name:">UQ darkgrey</td>
+      <td className="uqds-table__item" data-title="Name:">UQ Darkgrey</td>
```
Comment (uqmchu) [inline packages/storybook-html/src/components/core/colour-base.story.mdx:153]: Also revert this casing but in this case, add a space between dark and grey.

**Review by uqmchu — CHANGES_REQUESTED** [review]
I liked how you were proactive with systematising the neutral colours. However, the modified docs didn't need all those details. Also, see contrast issues with new alert colours.

File: packages/storybook-html/src/components/core/colour-apply.story.mdx (hunk 7 above)
Comment (uqtgeor4) [reply packages/storybook-html/src/components/core/colour-apply.story.mdx:305]: Removed

File: packages/storybook-html/src/components/core/colour-apply.story.mdx (hunk 6 above)
Comment (uqtgeor4) [reply packages/storybook-html/src/components/core/colour-apply.story.mdx:258]: Removed

File: packages/storybook-html/src/components/core/colour-apply.story.mdx (hunk 5 above)
Comment (uqtgeor4) [reply packages/storybook-html/src/components/core/colour-apply.story.mdx:184]: Removed

File: packages/storybook-html/src/components/core/colour-apply.story.mdx (hunk 4 above)
Comment (uqtgeor4) [reply packages/storybook-html/src/components/core/colour-apply.story.mdx:180]: Removed

File: packages/core/src/scss/_partials/_colour-constants.scss (hunk 3 above)
Comment (uqtgeor4) [reply packages/core/src/scss/_partials/_colour-constants.scss:8]: Don't know how that would have helped from renaming. Now that it's done this way, keeping it

File: packages/core/src/scss/_partials/_colour-constants.scss [hunk 10]
```diff
@@ -55,17 +79,19 @@ $link-hover: #005ea5;
 $link-active: $link-hover;
 
 // Alerts and Status - Dark
-$alert: $uq-red-dark20;
+$info: $uq-blue;
```
[the remark this answers is not in the document]
Comment (uqtgeor4) [reply packages/core/src/scss/_partials/_colour-constants.scss:82]: I have changed the brand blue to link blue

File: packages/core/src/scss/_partials/_colour-constants.scss (hunk 1 above)
Comment (uqtgeor4) [reply packages/core/src/scss/_partials/_colour-constants.scss:84]: Based on the function rules set, any darker is too dull and muddy. Not sure how to solve this. Or we may have to break the rule as an odd case

File: packages/storybook-html/src/components/core/colour-base.story.mdx (hunk 8 above)
Comment (uqtgeor4) [reply packages/storybook-html/src/components/core/colour-base.story.mdx:38]: Done

File: packages/storybook-html/src/components/core/colour-base.story.mdx (hunk 9 above)
Comment (uqtgeor4) [reply packages/storybook-html/src/components/core/colour-base.story.mdx:153]: Done

File: packages/core/src/scss/_partials/_colour-constants.scss (hunk 1 above)
Comment (uqmchu) [reply packages/core/src/scss/_partials/_colour-constants.scss:84]: It's fine that the yellow variant requires dark text. But make sure that the developer knows that.

**Review by uqmchu — APPROVED** [review]

**Pull request merged**